transformer_engine/pytorch/fp8.py

```diff
@@ -87,7 +87,13 @@ def get_amax_reduce_handle_fwd() -> Union[bool, None]:
 
 def get_global_fp8_buffer() -> Dict[str, List[torch.Tensor]]:
     """Returns global fp8 buffer."""
-    return _global_fp8_buffer
+    buffer = {}
+
+    # Map all tensors to CPU.
+    for k, v in _global_fp8_buffer.items():
+        buffer[k] = [tensor.cpu() for tensor in v]
+
+    return buffer
```

Comment:
This doesn't seem like the right place for checkpoint-specific logic. How about we change https://github.com/NVIDIA/TransformerEngine/blob/59be1028750be742e33a155d40306201d209bda5/transformer_engine/pytorch/module/base.py#L358
```python
state["global_fp8_buffer"] = {
    key: [tensor.cpu() for tensor in tensor_list]
    for key, tensor_list in _global_fp8_buffer.items()
}
```

Reply:
`get_global_fp8_buffer` is a checkpoint specific function. It's only use is to return the current global buffer for saving the checkpoint

Reply:
@timmoon10

Reply:
I see this implementation of `get_global_fp8_buffer` mirrors the implementation of `set_global_fp8_buffer`:

https://github.com/NVIDIA/TransformerEngine/blob/95ec156057d000c1ba5f26d3671e5ef7c149f763/transformer_engine/pytorch/fp8.py#L93-L101

That said, it's unintuitive. The function names and documentation don't make me think of checkpointing.